DebtBot/Services/BillService.cs

```diff
@@ -17,375 +17,394 @@
-    
+    {
+        var paymentUser = _userService.FindUser(parsedPayment.User, creator.Id);
+        paymentUser ??= dealWithUser(parsedPayment.User, creator);
+
+        var payment = _debtContext.BillPayments.FirstOrDefault(q => q.BillId == billId && q.UserId == paymentUser.Id);
+
+        if (payment is null)
+        {
+            payment = new BillPayment()
+            {
+                BillId = billId,
+                UserId = paymentUser.Id
+            };
+            _debtContext.BillPayments.Add(payment);
+        }
+
+        if (parsedPayment.Amount is not null)
+        {
+            payment.Amount = parsedPayment.Amount.Value;
+        }
+
+        _debtContext.SaveChanges();
+
+        //_publishEndpoint.Publish(new EnsureBillParticipant(billId, paymentUser.Id));
+        ensureBillParticipant(billId, paymentUser.Id);
+    }
+
     public bool Finalize(Guid id)
-    {
-		var bill = _debtContext.Bills.FirstOrDefault(q => q.Id == id);
-        if (bill == null)
-		{
+    {
+        var bill = _debtContext.Bills.FirstOrDefault(q => q.Id == id);
+        if (bill == null)
+        {
             return false;
+        }
+
+        if (bill.Status != ProcessingState.Draft)
+        {
+            return false;
         }
 
-		if(bill.Status != ProcessingState.Draft)
-		{
-			return false;
-		}
-
         bill.Status = ProcessingState.Ready;
         _debtContext.SaveChanges();
 
-        _publishEndpoint.Publish(new BillFinalized(id));
-
-		return true;
+        _publishEndpoint.Publish(new BillFinalized(id));
+
+        return true;
     }
 
     public bool HasAccess(Guid userId, BillModel? bill)
-    {
-	    if (bill is null)
-	    {
-		    return false;
-	    }
-	    
-	    if (bill.Creator.Id == userId)
-	    {
-		    return true;
-	    }
-	    
-	    if(bill.Payments.Any(q => q.User.Id == userId))
-	    {
-		    return true;
-	    }
-
-	    if(bill.Lines.Any(q => q.Participants.Any(w => w.User.Id == userId)))
-	    {
-		    return true;
-	    }
-
-	    return false;
+    {
+        if (bill is null)
+        {
+            return false;
+        }
+
+        if (bill.Creator.Id == userId)
+        {
+            return true;
+        }
+
+        if (bill.Payments.Any(q => q.User.Id == userId))
+        {
+            return true;
+        }
+
+        if (bill.Lines.Any(q => q.Participants.Any(w => w.User.Id == userId)))
+        {
+            return true;
+        }
+
+        return false;
     }
 
     public List<BillModel> GetByUser(Guid userId)
-    {
+    {
+
         var bills = _debtContext
             .Bills
             .Include(q => q.Lines)
             .ThenInclude(q => q.Participants)
             .ThenInclude(q => q.User)
             .Include(q => q.Payments)
             .ThenInclude(q => q.User)
-            .Where(q => q.CreatorId == userId)
+            .Where(q => q.CreatorId == userId || q.Payments.Any(w => w.UserId == userId) || q.Lines.Any(w => w.Participants.Any(e => e.UserId == userId)))
```

Comment:
this is the method to get bill Created by User
did you mix it up with GetForUser which already have 
`			.Bills
			.Where(b => b
				.BillParticipants
				.Any(p => p.UserId == userId))`